python/ray/_private/usage/usage_lib.py

```diff
@@ -525,29 +564,42 @@ def get_library_usages_to_report(gcs_client) -> List[str]:
         return []
 
 
-def _parse_extra_usage_tags() -> Dict[str, str]:
-    """Parse the extra usage tags given by the environment variable.
+def get_extra_usage_tags_to_report(gcs_client) -> Dict[str, str]:
+    """Get the extra usage tags from env var and gcs kv store.
 
     The env var should be given this way; key=value;key=value.
     If parsing is failed, it will return the empty data.
 
     Returns:
-        Dictionary of key value pair parsed.
+        Extra usage tags as kv pairs.
     """
-    extra_tags = os.getenv("RAY_USAGE_STATS_EXTRA_TAGS", None)
-    if not extra_tags:
-        return None
+    extra_usage_tags = dict()
+
+    extra_usage_tags_env_var = os.getenv("RAY_USAGE_STATS_EXTRA_TAGS", None)
+    if extra_usage_tags_env_var:
+        try:
+            kvs = extra_usage_tags_env_var.strip(";").split(";")
+            for kv in kvs:
+                k, v = kv.split("=")
+                extra_usage_tags[k] = v
+        except Exception as e:
+            logger.info(f"Failed to parse extra usage tags env var. Error: {e}")
 
     try:
-        result = {}
-        kvs = extra_tags.strip(";").split(";")
-        for kv in kvs:
-            k, v = kv.split("=")
-            result[k] = v
-        return result
+        keys = gcs_client.internal_kv_keys(
+            usage_constant.EXTRA_USAGE_TAG_PREFIX.encode(),
+            namespace=usage_constant.USAGE_STATS_NAMESPACE.encode(),
+        )
+        for key in keys:
```

Comment:
If perf becomes an issue: we can create a new kv api to return both keys and values for a prefix